### PR TITLE
12.0 mig sale_timesheet_invoice_description

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please comment when Travis is green.

**Comment by ernestotejeda** [comment]
Travis in green

File: sale_timesheet_invoice_description/models/sale.py [hunk 1]
```diff
@@ -29,7 +29,12 @@ class SaleOrderLine(models.Model):
     def _prepare_invoice_line_details(self, line, desc_rule):
         details = []
         if desc_rule[0] == '1':
-            details.append(line.date)
+            # If project_timesheet_time_control module is installed,
+            # date field is hidden and the new date_time field is used instead
+            if hasattr(line, 'date_time'): # pragma: no cover
```
Comment (pedrobaeza) [inline sale_timesheet_invoice_description/models/sale.py]: Although the field is hidden, `date` field is still synchronized, so I prefer to keep the previous behavior showing only date. Customer doesn't need this kind of detail.

File: sale_timesheet_invoice_description/models/sale.py [hunk 2]
```diff
@@ -47,9 +52,14 @@ def _prepare_invoice_line(self, qty):
...
-        for line in self.env['account.analytic.line'].search(
-                domain, order='date, id'):
+            last_date = fields.Datetime.to_string(last_invoice.create_date)
+            domain.append(('create_date', '>', last_date))
+
+        # If project_timesheet_time_control module is installed,
+        # date field is hidden and the new date_time field is used instead
+        line_obj = self.env['account.analytic.line']
+        order = hasattr(line_obj, 'date_time') and 'date_time' or 'date'
+        for line in line_obj.search(domain, order=order + ', id'):
```
Comment (pedrobaeza) [inline sale_timesheet_invoice_description/models/sale.py]: Why don't rely in default order set by the model instead (and more now that we are going to set it properly)?

File: sale_timesheet_invoice_description/models/sale.py (hunk 1 above)
Comment (ernestotejeda) [reply sale_timesheet_invoice_description/models/sale.py]: Yes, I agree, thanks

File: sale_timesheet_invoice_description/models/sale.py (hunk 2 above)
Comment (ernestotejeda) [reply sale_timesheet_invoice_description/models/sale.py]: I think you're right, thank you

**Review by pedrobaeza — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]
Tested in runbot :+1:

**Pull request merged**